### PR TITLE
Add script for running gradio demo from a local checkpoint

### PR DESCRIPTION
**Pull request by coryMosaicML**

### CODE REVIEW

File: diffusion/inference/inference_model.py [hunk 1]
```diff
@@ -51,7 +53,7 @@ def __init__(self,
         )
 
         if not pretrained:
-            state_dict = torch.load(LOCAL_CHECKPOINT_PATH)
+            state_dict = torch.load(local_checkpoint_path)
             for key in list(state_dict['state']['model'].keys()):
```
Comment (Skylion007) [inline diffusion/inference/inference_model.py:57]:
```suggeestion
            for key in state_dict['state']['model'].keys():
```

File: scripts/gradio_demo.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+parser.add_argument('--resolution', default=512, type=int, help='Resolution to generate images at.')
+parser.add_argument('--share', default=False, action='store_true', help='Share the demo publicly.')
+parser.add_argument('--progress_bar', default=False, action='store_true', help='Show progress bar.')
+args = parser.parse_args()
+
+
+class ImageGenerator:
+    """Wrapper class to interface model deployment with Gradio."""
+
+    def __init__(self, model_deployment, resolution, progress_bar):
```
Comment (Skylion007) [inline scripts/gradio_demo.py:27]: nit: typehints

**Review by A-Jacobson — APPROVED** [review]
Approved bc useful + not breaking, but would like to see some mention that this exists and how to use it in the README.

File: scripts/gradio_demo.py (hunk 2 above)
Comment (coryMosaicML) [reply scripts/gradio_demo.py]: Added!

**Pull request merged**